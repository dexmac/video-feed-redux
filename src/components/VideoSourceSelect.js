import React from "react";
import PropTypes from "prop-types";
import * as constants from "../constants/constants";
import Select from "react-select";
import "react-select/dist/react-select.css";

const VideoSourceSelect = ({ selectedValues, actionOnSelection }) => (
  <Select
    name="video-source-select"
    multi
    value={selectedValues}
    placeholder={constants.SELECT_VIDEO_SOURCE_PLACEHOLDER}
    onChange={actionOnSelection}
    options={[
      {
        value: constants.URL_VIDEO_SOURCE,
        label: constants.URL_VIDEO_SOURCE
      },
      {
        value: constants.YOUTUBE_VIDEO_SOURCE,
        label: constants.YOUTUBE_VIDEO_SOURCE
      },
      {
        value: constants.FACEBOOK_VIDEO_SOURCE,
        label: constants.FACEBOOK_VIDEO_SOURCE
      }
    ]}
  />
);

VideoSourceSelect.propTypes = {
  selectedValues: PropTypes.array,
  actionOnSelection: PropTypes.func.isRequired // onchange handler, supplied by feed manager
};

export default VideoSourceSelect;
