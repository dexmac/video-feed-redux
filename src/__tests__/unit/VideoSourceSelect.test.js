import React from "react";
import jest from "jest-mock";
import VideoSourceSelect from "../../components/VideoSourceSelect";

describe("VideoSourceSelect", () => {
  let props;
  let mountedVideoSourceSelect;

  /**
   * Clearing props before each test
   */
  beforeEach(() => {
    props = {
      selectedValues: undefined,
      actionOnSelection: undefined
    };
    mountedVideoSourceSelect = undefined;
  });

  /**
   * Shallow rendering vs. snapshot test
   */
  it("renders correctly && matches snapshot", () => {
    const actionOnSelectionMock = jest.fn();
    const tree = shallow(
      <VideoSourceSelect
        selectedValues={[]}
        actionOnSelection={actionOnSelectionMock}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
