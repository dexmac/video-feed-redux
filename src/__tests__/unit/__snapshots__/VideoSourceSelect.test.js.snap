// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoSourceSelect renders correctly && matches snapshot 1`] = `
<Select
  arrowRenderer={[Function]}
  autosize={true}
  backspaceRemoves={true}
  backspaceToRemoveMessage="Press backspace to remove {label}"
  clearAllText="Clear all"
  clearRenderer={[Function]}
  clearValueText="Clear value"
  clearable={true}
  closeOnSelect={true}
  deleteRemoves={true}
  delimiter=","
  disabled={false}
  escapeClearsValue={true}
  filterOptions={[Function]}
  ignoreAccents={true}
  ignoreCase={true}
  inputProps={Object {}}
  isLoading={false}
  joinValues={false}
  labelKey="label"
  matchPos="any"
  matchProp="any"
  menuBuffer={0}
  menuRenderer={[Function]}
  multi={true}
  name="video-source-select"
  noResultsText="No results found"
  onBlurResetsInput={true}
  onChange={[Function]}
  onCloseResetsInput={true}
  onSelectResetsInput={true}
  openOnClick={true}
  optionComponent={[Function]}
  options={
    Array [
      Object {
        "label": "url",
        "value": "url",
      },
      Object {
        "label": "youtube",
        "value": "youtube",
      },
      Object {
        "label": "facebook",
        "value": "facebook",
      },
    ]
  }
  pageSize={5}
  placeholder="Sources to filter by..."
  removeSelected={true}
  required={false}
  rtl={false}
  scrollMenuIntoView={true}
  searchable={true}
  simpleValue={false}
  tabSelectsValue={true}
  trimFilter={true}
  value={Array []}
  valueComponent={[Function]}
  valueKey="value"
/>
`;
